valhalla/baldr/graphreader.h

```diff
@@ -947,6 +947,15 @@ class GraphReader {
    */
   int GetTimezone(const baldr::GraphId& node, graph_tile_ptr& tile);
 
+  /**
+   * Convenience method to get the timezone index from an edge. Preferably it returns
+   * the start's node's timezone.
+   * @param edge   GraphId of the edge to get the timezone index.
+   * @param tile   Current tile.
+   * @return Returns the timezone index. A value of 0 indicates an invalid timezone.
+   */
+  int GetTimezoneFromEdge(const baldr::GraphId& edge, graph_tile_ptr& tile);
```

Comment:
just need to check and mention if the tile is modified. if its not modified we should pass it wihtout the reference if modified we should say whether its for the end or begin node of the edge

Reply:
you're right, in `nodeinfo()` the tile can change, so we could end up with the edge's end node's tile instead. is a comment in the docstring good enough for you or you want smth more elaborate?

Reply:
yeah just docstring update was all i wanted. i hope that its a gaurantee though that as a post condition it will be the endnodes tile.

Reply:
Fwiw, our own usage of this (so far) is safe, we don’t even use the tile, only for caching

Reply:
yep yep its just to avoid mistakes

Reply:
> i hope that its a gaurantee though that as a post condition it will be the endnodes tile not just could be, i guess when the tile isnt found it could be, thats ok.

after this function is called, it's very unlikely that the tile is changed (unless you compiled the graph without timezone info, but maybe that's even handled further up in the processing chain). that only happens if the edge's start node has no timezone info.